tutorials/08-implementing-a-custom-feature.md

```diff
@@ -0,0 +1,274 @@
+\ref ignition::physics::Feature "Feature".
+
+As defined, the `RetrieveWorld` feature retrieves
+world pointer from physics engine, so it is natural to define `World` entity inherited
+from \ref ignition::physics::Feature::World "Feature::World" and declare the
+necessary member function `GetDartsimWorld`. Then we define the `Implementation`
+class having virtual member function for overriding in the actual implementation of
+the custom feature `RetrieveWorld` later.
+
+Finally, we implement the `World`
+entity's member function `GetDartsimWorld` to call the `Implementation`
+class's member function `GetDartsimWorld` via
+\ref ignition::physics::Feature::Entity::Interface "Entity::Interface"
+convenience function for querying the feature `Implementation` object.
+
+### Implement the custom feature
+
+After defining the custom feature, please look into where it is added to a
+\ref ignition::physics::FeatureList "FeatureList" in
+[dartsim/src/CustomFeatures.hh](https://github.com/ignitionrobotics/ign-physics/blob/main/dartsim/src/CustomFeatures.hh)
+and implemented in [dartsim/src/CustomFeatures.cc](https://github.com/ignitionrobotics/ign-physics/blob/main/dartsim/src/CustomFeatures.cc).
+We display them here for convenience:
+
+- `CustomFeatures.hh`:
+
+```cpp
+#include <ignition/physics/Implements.hh>
+#include <ignition/physics/dartsim/World.hh>
+#include "Base.hh"
+
+namespace ignition {
+namespace physics {
+namespace dartsim {
+
+using CustomFeatureList = FeatureList<
+  RetrieveWorld
+>;
+
+class CustomFeatures :
+    public virtual Base,
+    public virtual Implements3d<CustomFeatureList>
+{
+  public: dart::simulation::WorldPtr GetDartsimWorld(
+      const Identity &_worldID) override;
+};
+
+}
+}
+}
+
+```
+
+The custom feature `RetrieveWorld` is added to `CustomFeatureList`, other custom
+features could also be added here.
+The `CustomFeatures` "FeatureList" here inherits from:
+- [Base](https://github.com/ignitionrobotics/ign-physics/blob/main/dartsim/src/Base.hh)
+class for foundation definitions of Models, Joints, Links, and Shapes objects
+of `dartsim` interfacing to Ignition Physics API.
+- \ref ignition::physics::Implements3d "Implements3d" for implementing the
+custom feature with \ref ignition::physics::FeaturePolicy3d "FeaturePolicy3d"
+("FeaturePolicy" of 3 dimensions and scalar type `double`).
+
+Based on the `CustomFeatureList`, we will override the corresponding member functions
+declared in each of the custom features. In this case, we have only the custom
+feature `RetrieveWorld` and its corresponding `Implementation::GetDartsimWorld`
+member function.
+
+- `CustomFeatures.cc`:
+
+```cpp
+#include "CustomFeatures.hh"
+
+namespace ignition {
+namespace physics {
+namespace dartsim {
+
+/////////////////////////////////////////////////
+dart::simulation::WorldPtr CustomFeatures::GetDartsimWorld(
+    const Identity &_worldID)
+{
+  return this->worlds.at(_worldID);
+}
+
+}
+}
+}
+```
+
```

Comment:
At this point, it'd be helpful to show what files are included in this particular plugin folder, just as a sanity check for users to follow along, before they proceed to compile and test the new feature.